modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,134 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+== Functions
+
+The Rust SDK includes the following https://doc.rust-lang.org/rust-by-example/fn.html[function] to manage data transforms to an output topic.  
+
+=== on_record_written
+
+[,rust]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+The `on_record_written()` function transforms individual records after they have been written to an input topic. The function registers a callback to be fired when a record is written to the topic. This callback is triggered after the record has been written and fsynced to disk and the producer has been acknowledged (see <<writeevent,`WriteEvent`>>).
```

Comment:
```suggestion
The `on_record_written()` function registers a callback that transforms records after they have been written to an input topic. This function is executed when a record is written to the topic. The callback is executed after the record has been written and fsynced to disk and the producer has been acknowledged.

Callback parameters

- <<writeevent,`WriteEvent`>>
- <<recordwriter, `RecordWriter`>>

Example:

This example transform function copies the same data from an input topic to an output topic.:

[,rust]
----
use anyhow::Result;
use redpanda_transform_sdk::*;

fn main() {
    on_record_written(my_transform);
}

// This will be called for each record in the source topic. The output records returned will be written to the destination topic.
fn my_transform(event: WriteEvent, writer: &mut RecordWriter) -> Result<()> {
    writer.write(event.record)?;
    Ok(())
----
```

Maybe we should detail what happens to errors? Does the error interrupt the writing process?

Reply:
Might also be nice to show error handling in the example

Reply:
@rockwotj Would you be able to provide more details on these?

Reply:
There is "error handling" in that if `write` fails we return an error. That's what the `?` does. I don't think it's expected that users handle errors in this example.

If an error is returned, we end up printing an error and aborting the "process" in the VM, and then Redpanda will retry processing with backoff.